Remove THCNumerics

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66391
* #66480
* #66390
* #66389
* __->__ #66388

Differential Revision: [D31547710](https://our.internmc.facebook.com/intern/diff/D31547710)

peterbell10 committed:
Remove THCNumerics

[ghstack-poisoned]

ngimel reviewed (approved):
So by far the most used thing from THCNumerics was `isnan` :-) ?

peterbell10 committed:
Update on "Remove THCNumerics"


Differential Revision: [D31547710](https://our.internmc.facebook.com/intern/diff/D31547710)

[ghstack-poisoned]
